pandas/tests/arithmetic/test_timedelta64.py

```diff
@@ -1761,6 +1763,8 @@ def test_td64arr_floordiv_td64arr_with_nat(self, box_with_array):
 
         expected = np.array([1.0, 1.0, np.nan], dtype=np.float64)
         expected = tm.box_expected(expected, xbox)
+        if box is DataFrame and using_array_manager:
+            expected[[0, 1]] = expected[[0, 1]].astype("int64")
```

Comment:
why is expected different and is one behavior more desirable than the other?

Reply:
ArrayManager performs the op column-wise, and thus can preserve dtypes (floordiv gives ints). Will add a comment about it in the line of https://github.com/pandas-dev/pandas/pull/39753/files#r576350947

Reply:
thanks